vuu/src/test/scala/org/finos/vuu/core/filter/FilterClauseTest.scala

```diff
@@ -0,0 +1,60 @@
+package org.finos.vuu.core.filter
+
+import org.finos.vuu.core.table.RowWithData
+import org.scalatest.featurespec.AnyFeatureSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks._
+
+class FilterClauseTest extends AnyFeatureSpec with Matchers {
+
+  private val testRow = RowWithData("Key", Map(
+    "null-col" -> null,
+    "ric" -> "TEST.L",
+    "size" -> 4,
+    "timestamp" -> 5L,
+    "priceInFloat" -> 100f,
+    "priceInDouble" -> 100.0d,
+    "true-col" -> true,
+    "false-col" -> false
+  ))
+
+  Feature("EqualsClause.filter") {
+
+    forAll(Table(
+      ("columnType", "columnName", "value", "expected"),
+      ("String", "ric", "TEST.L", true),
+      ("String", "ric", "TEST.NOT-EQ", false),
+      ("Int", "size", "4", true),
+      ("Int", "size", "7", false),
+      ("Long", "timestamp", "5", true),
+      ("Long", "timestamp", "0", false),
+      ("Float", "priceInFloat", "100.0", true),
+      ("Float", "priceInFloat", "100.1", false),
+      ("Double", "priceInDouble", "100.0", true),
+      ("Double", "priceInDouble", "100.1", false),
+      ("Boolean", "true-col", "true", true),
+      ("Boolean", "true-col", "false", false),
+      ("Null", "null-col", "null", false)
```

Comment:
I think there is two more edge cases here
1. if the there is a column with value null and we call filter(null) i would expect it to return true.
2. If there is a column that is no in the test data and you try to filter for that column i guess we need to report that as error.. not sure what it does currently 

The second one i would write its own test as its edge cases